### PR TITLE
timestop the SM

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request

Allows the timestop effects to freeze the Supermatter processing so that engineers might be able to save it this time
## Why It's Good For The Game

Both wizards and science nerds are able to stop the SM seconds from meltdown so that engies can figure out what's the right pipe to place to prevent a destruction of property.
## Changelog
:cl:
add: Timestop blocks the SM from processing (can't be healed, doesn't output gases but can still take damage)
/:cl:

### CODE REVIEW

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -235,7 +237,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 /obj/machinery/power/supermatter_crystal/process_atmos()
 	// PART 1: PRELIMINARIES
-	if(disable_process)
+	if(disable_process || time_frozen)
```
Comment (vincentiusvin) [inline code/modules/power/supermatter/supermatter.dm:240]: If you are just going to replicate what disable process does, consider making it a var that can hold 3 values (defines):

- Not disabled
- Disabled in game (if check in your unfreeze supermatter code, set it back to none if its this)
- Disabled fully (unfreeze supermatter cant unfreeze it if its set to this, set hugbox supermatters to this)

You can look into the defines i use in sm delam priority for an example.



If you make a new variable for every single little feature sm will get ugly again :(

**Review by MrMelbert — COMMENTED** [review]
Saying what the signal was created for is pointless, they're supposed to be generic events. 
Instead mention what arguments are passed.

File: code/datums/proximity_monitor/fields/timestop.dm [hunk 2]
```diff
@@ -97,6 +97,7 @@
 	freeze_atom(movable)
 
 /datum/proximity_monitor/advanced/timestop/proc/freeze_atom(atom/movable/A)
+	SIGNAL_HANDLER
```
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/datums/proximity_monitor/fields/timestop.dm:100]: yeah, i remembered wrong the use of the define (inverted the use)

**Comment by Overuseddart** [comment]
So if I get 9 monkeyhuman/golems under my control turned into slimefolk, and told them all to space out timestops, could I theoretically make a un-dellamable SM?

File: code/datums/proximity_monitor/fields/timestop.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/datums/proximity_monitor/fields/timestop.dm:100]: Yep the commit didn't get pushed, I'll do it later sorry about that

**Pull request merged**